src/contributing.md

```diff
@@ -0,0 +1,140 @@
+
+Ensure proper Keyboard support and focus-scope handling.
+
+- For Views, Panels ensure proper tab behavior [see `KeyNavigation`
+](https://doc.qt.io/qt-6/qml-qtquick-keynavigation.html)
+- For keyboard input item, ensure focus properties are set [See Qt docs](https://doc.qt.io/qt-6/qtquick-input-focus.html)
+
+When unsure, check Qt's excellent documentation.
+
+- For main components/controls, there is a good overview that is worth revisiting from time to time
+- Functions/Properties usually have a short on-point description
+- In QtCreator you can quickly open the doc panel using `F1` key while having a cursor on component or function/property and recheck its invariants, pre and post conditions
+
+Consider that design follow user-friendlier principles.
+
+- Have commonly used items at hand
+- Have transition if possible
+  - Use Qt's property animation for states
+- Avoid often and radical size changes to items in views/layouts. Radical size change confuses users.
+  - If the content is not available, consider having placeholders using the estimated size (delegates, dummy items)
+
+## QML well-behaved components checklist
+
+### Sizing support
+
+- Define size hints appropriately. Define implicit properties (`impicitWidth` and `implicitHeight`)
+  - They are used to break polishing binding loops for adaptive control
+  - All the containers use the implicit size for deriving the initial size (Layouts, Popups, Delegates, GridView, ListView ...). Size hints combined with resize adaptability are the appropriate way to have reactive controls.
+  - For complex controls, look if layouts are the choice as the main position driver. [Layouts](###Layouts)
+
+  ```qml
+  Item {
+    id: root
+    implicitWidth: mainLayout.implicitWidth
+    implicitHeight: mainLayout.implicitHeight
+    RowLayout {	// Column, Grid
+       id: mainLayout
+       // ...
+    }
+  ```
+
+- Define a default size to work with positioners (`Row`, `Column`, `Grid`). If appropriate, source them from implicit dimensions and expect the owner to overwrite them.
+  - Don't mix hints with sizes, it will create binding loops
+- Adapt to the requested size.
+Have the base control as the component's root. This way, control inherits all the interface and reduce the code.
+If it doesn't map to an existing one, use `Item` (never use layout or positioners as base control, they don't behave as expected)
+Have public properties at the top and make them explicit
+
+## Topics
+
+### Layouts
+
+Hierarchically sources hints from children. Implicit properties will have the recommended aggregated size.
+
+Layouts as children of other layouts have `fillWidth` and `fillHeight` enabled; controls don't.
+
+Use `Layout.prefferedWidth` and `Layout.prefferedHeight` attached properties to overwrite item's implicitWidth if they are not appropriate to the design.
+
+### Scope
+
+Component 'id's are not accessible outside component's scope
```

Comment:
Maybe it would be worth mentioning the term  `dynamic scoping` here.

> Dynamic scoping is very powerful, but it must be used cautiously to prevent the behavior of QML code from becoming difficult to predict. In general it should only be used in cases where the two components are already tightly coupled in another way. When building reusable components, it is preferable to use property interfaces.
https://doc.qt.io/qt-5/qtqml-documents-scope.html#component-instance-hierarchy

Reply:
Done, thanks for the suggestion